src/algokit/cli/tasks/analyze.py

```diff
@@ -0,0 +1,226 @@
+)
+@click.argument(
+    "input_paths",
+    nargs=-1,
+    type=click.Path(exists=True, dir_okay=True, file_okay=True, path_type=Path),
+    required=True,
+)
+@click.option(
+    "-r",
+    "--recursive",
+    is_flag=True,
+    cls=MutuallyExclusiveOption,
+    not_required_if=["file"],
+    help="Recursively search for all TEAL files within the provided directory.",
+)
+@click.option(
+    "--force",
+    is_flag=True,
+    help="Force verification without the disclaimer confirmation prompt.",
+)
+@click.option(
+    "-b",
+    "--baseline",
+    is_flag=True,
+    help=(
+        "Exit with a non-zero code if any diffs are identified between the current "
+        "and baseline reports. By default baseline reports are stored in the "
+        ".algokit/static-analysis/artifacts folder. Alternatively, you can specify a "
+        "custom path using the --output option to compare against."
+    ),
+)
+@click.option(
+    "-o",
+    "--output",
+    "output_path",
+    required=False,
+    default=None,
+    type=click.Path(dir_okay=True, file_okay=False, resolve_path=True, path_type=Path),
+    help="Directory path where to store the results of the static analysis.",
+)
+@click.option(
+    "-e",
+    "--exclude",
+    "detectors_to_exclude",
+    multiple=True,
+    default=[],
+    type=click.STRING,
+    help="Exclude specific vulnerabilities from the analysis. Supports multiple exclusions in a single run.",
+)
+def analyze(  # noqa: PLR0913
+    *,
+    input_paths: tuple[Path],
+    recursive: bool,
+    force: bool,
+    baseline: bool,
+    output_path: Path | None,
+    detectors_to_exclude: list[str],
+) -> None:
+    """
+    Analyze the TEAL programs for common vulnerabilities.
+    """
+
+    prepare_artifact_folders(output_path)
+    detectors_to_exclude = sorted(set(detectors_to_exclude))
+    input_files = get_input_files(input_paths=input_paths, recursive=recursive)
+
+    if not force:
+        click.confirm(
+            "Standard disclaimer: This tool provides suggestions for improving "
+            "your TEAL programs, but it does not guarantee their correctness "
+            "or security. Do you understand?",
+            abort=True,
+        )
+
+    reports = {}
+    for cur_file in input_files:
+        file = cur_file.resolve()
+
+        if has_template_vars(file):
+            click.secho(
+                f"Warning: skipping {file} due to template variables. Substitute them before scanning.",
+                err=True,
+                fg="yellow",
+            )
+            continue
+
+        filename = generate_report_filename(file)
+
+        # If baseline is enabled, store the report in the snapshots folder otherwise store it in the reports folder
+        # If a custom output path is provided, store the report in the specified path
+        report_output_root = TEALER_SNAPSHOTS_ROOT if baseline else output_path or TEALER_REPORTS_ROOT
```

Comment:
Is `TEALER_SNAPSHOTS_ROOT` used for anything other than baseline reports? Potentially `TEALER_BASELINE_ROOT` is a more consistent name?

Reply:
Can still rename it back, but now its the main path that i used for persisting the reports